internal/rsl/rsl_test.go

```diff
@@ -395,6 +395,50 @@ func TestGetLatestReferenceEntry(t *testing.T) {
 		assert.ErrorIs(t, err, ErrRSLEntryNotFound)
 	})
 
+	t.Run("with ref name and invalid before entry id", func(t *testing.T) {
+		tempDir := t.TempDir()
+		repo := gitinterface.CreateTestGitRepository(t, tempDir, false)
+		// RSL structure for the test
+		// 1    <- 2
+		// main <- feature
+		testRefs := []string{"main", "feature"}
+		for _, ref := range testRefs {
+			if err := NewReferenceEntry(ref, gitinterface.ZeroHash).Commit(repo, false); err != nil {
+				t.Fatal(err)
+			}
+		}
+		hash, hashErr := gitinterface.NewHash("a94a8fe5ccb19ba61c4c0873d391e987982fbbd3")
+		entry, annotations, err := GetLatestReferenceEntry(repo, ForReference("main"), BeforeEntryID(hash), UntilEntryNumber(0))
+		assert.Empty(t, hashErr)
```

Comment:
side note: Empty is the wrong one fwiw. For errors, we want to assert Nil if we don't expect an error.

Reply:
Understood, addressed